core/src/main/java/io/smallrye/openapi/runtime/scanner/spi/AnnotationScanner.java

```diff
@@ -466,87 +466,126 @@ default void createResponseFromRestMethod(final AnnotationScannerContext context
             final MethodInfo method,
             Operation operation) {
 
-        Type returnType = context.getResourceTypeResolver().resolve(method.returnType());
-        APIResponse response = null;
+        final Type returnType = context.getResourceTypeResolver().resolve(method.returnType());
+        final boolean internalResponse = isScannerInternalResponse(returnType, context, method);
         final int status = getDefaultStatus(method);
-        final String code = String.valueOf(status);
-        final String description = getReasonPhrase(status);
+        final String code;
+        final String description;
 
-        if (isVoidResponse(method)) {
-            if (generateResponse(code, operation)) {
-                response = OASFactory.createAPIResponse().description(description);
+        if (internalResponse && context.getConfig().genericResponseUseDefault()) {
+            code = "default";
+            description = "Default Response";
+        } else {
+            code = String.valueOf(status);
+            description = getReasonPhrase(status);
+        }
+
+        if (!generateResponse(code, operation)) {
+            return;
+        }
```

Comment:
It occurred to me as a potential problem that if an operation had a `200` response specified for the operation already (e.g. from a static file), and we have the new `generic-response-use-default` option set, would this result in us generating a `default` response in addition to the existing `200` response?

We would call `generateResponse("default", operation)`, which would return `true` because there's no `default` response specified yet.